### PR TITLE
System tests for API and API Impl (Native and JS)

### PR DESCRIPTION
**Pull request by gmbharath12**
- System Tests for API and API implementation (both Native and JS)
- https://github.com/electrode-io/electrode-native/issues/491

### CODE REVIEW

**Comment by krunalsshah** [comment]
@gmbharath12 Thanks for test.

system-tests-fixtures/api/complex-api/schema.json - The schema is on similar lines with the system-tests-fixtures/default-schema/schema.json , I am not sure what complexity is different in the two. If there is a delta of complexity between either of the 2, just use the 1 which is more complex

**Comment by krunalsshah** [comment]
@gmbharath12 The 2 files are duplicate of each other. I may be missing a point, just wanted to confirm if it was unintentional :)
system-tests-fixtures/default-schema/schema.json
system-tests-fixtures/api/default-api/TestApi/schema.json

**Comment by gmbharath12** [comment]
@krunalsshah 
system-tests-fixtures/default-schema/schema.json ----- Used this schema to generate fixture. In future if anyone wants to use the schema can use this schema. 
system-tests-fixtures/api/default-api/TestApi/schema.json -- it's a test fixture schema that was generated by command.

**Comment by belemaire** [comment]
@gmbharath12 
As discussed, let's add a more complex API schema test case covering most of the constructs offered by our API generators, but overall looks good to me.
That being said, please do not issue WIP PRs (that are failing tests) or PRs with "Do not merge". 
In the future, in case you want a pre-PR review, please just link me your branch ;)

**Comment by belemaire** [comment]
By the way I also think we can get rid of the nested structure `default-api/TestApi` (similar for other API fixtures). Let's just name this API `DefaultApi` for example, as `default-api` contains nothing more than the API folder, it does not really makes sense.

**Comment by gmbharath12** [comment]
@belemaire Sure 👍

File: system-tests.js [hunk 1]
```diff
@@ -238,4 +242,28 @@ run('ern platform list')
...
 run('ern platform plugins search react-native')
 
-afterAll()
+afterAll() */
+
+// System Tests for API
+console.log(`#### Path to Default schema ${pathToDefaultSchema}`)
+console.log(`#### Path to Default API ${pathToDefaultAPI}`)
+console.log(`#### Path to generated API ${workingDirectoryPath}`)
+run(`ern create-api ${apiName} -p ${apiPkgName} --skipNpmCheck --schemaPath ${pathToDefaultSchema}`)
```
Comment (krunalsshah) [inline system-tests.js]: can we please adapt the existing api test to work with folder content comparison? To limit the test run time

File: system-tests.js [hunk 2]
```diff
@@ -238,4 +242,28 @@ run('ern platform list')
...
+assert(areSameDirectoriesContent(pathToDefaultAPI, workingDirectoryPath), 'Generated API differ from reference fixture !')
+console.log('Done, test passes API !!!!!!!!!')
+
+// Tests for API IMPL Native
+console.log(`### Path to Native-impl ${pathToApiImplNative}`)
+console.log(`### Path to workingDirectoryPath ${workingDirectoryPath}`)
+process.chdir(workingDirectoryPath)
+run(`ern create-api-impl ${movieApi} -p ${movieApiImplPkgName} --skipNpmCheck --nativeOnly --outputDirectory ${workingDirectoryPath} --force`)
```
Comment (krunalsshah) [inline system-tests.js]: can we please adapt the existing api-impl test to work with folder content comparison? To limit the test run time

File: system-tests.js [hunk 3]
```diff
@@ -117,11 +122,10 @@ function areSameDirectoriesContent (pathA, pathB, filesToIgnoreContentDiff) {
   }
   return result
 }
-
 console.log(info(`Entering temporary working directory : ${workingDirectoryPath}`))
 process.chdir(workingDirectoryPath)
 
-console.log(info(`Creating GitHub repository (${gitHubCauldronRepositoryName})`))
+/* console.log(info(`Creating GitHub repository (${gitHubCauldronRepositoryName})`))
```
Comment (krunalsshah) [inline system-tests.js]: accidental comment

File: system-tests-fixtures/default-schema/schema.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
+{
```
Comment (krunalsshah) [inline system-tests-fixtures/default-schema/schema.json]: looks the output of this schema is generated in `TestApi` folder? Can you please keep the folder name consistent i.e. either call the folder as `DefaultApi` or rename this folder to `test-schema`?

File: system-tests-fixtures/api/complex-api/schema.json [hunk 5]
```diff
@@ -0,0 +1,117 @@
+{
```
Comment (krunalsshah) [inline system-tests-fixtures/api/complex-api/schema.json]: is this schema used yet?

Comment (gmbharath12) [reply system-tests-fixtures/api/complex-api/schema.json]: @krunalsshah I'm updating this PR.. So a lot will be changed. Please review it again once I push those changes in a bit.

**Pull request merged**